router/src/server.rs

```diff
@@ -1537,70 +1548,44 @@ pub async fn run(
         .route("/v1/completions", post(completions_v1))
         .route("/v1/chat/completions", post(chat_completions_v1))
         // AWS Sagemaker route
-        .route("/invocations", post(compat_generate))
+        .route("/invocations", post(compat_generate));
+
+    let info_routes = Router::new()
+        .route("/", get(health))
         // Base Health route
         .route("/health", get(health))
-        // Inference API health route
-        .route("/", get(health))
+        .route("/info", get(get_model_info))
         // AWS Sagemaker health route
         .route("/ping", get(health))
         // Prometheus metrics route
         .route("/metrics", get(metrics))
-        .route("/tokenize", post(tokenize))
+        .route("/tokenize", post(tokenize));
+
+    // Combine routes and layers
+    let mut app = Router::new()
+        .merge(swagger_ui)
+        .merge(base_routes)
+        .merge(info_routes);
+
+    // add layers after routes
+    app = app
         .layer(Extension(info))
         .layer(Extension(client.clone()))
         .layer(Extension(request_logger_sender.clone()))
         .layer(Extension(health_ext.clone()))
         .layer(Extension(compat_return_full_text))
         .layer(Extension(infer))
         .layer(Extension(prom_handle.clone()))
-        .layer(opentelemetry_tracing_layer())
+        .layer(OtelAxumLayer::default())
         .layer(cors_layer)
         .layer(Extension(cloned_tokenizer));
 
     if ngrok {
         #[cfg(feature = "ngrok")]
         {
-            use ngrok::config::TunnelBuilder;
-
-            let _ = addr;
-
-            let authtoken =
-                ngrok_authtoken.expect("`ngrok-authtoken` must be set when using ngrok tunneling");
-
-            let edge = ngrok_edge.expect("`ngrok-edge` must be set when using ngrok tunneling");
-
-            let tunnel = ngrok::Session::builder()
-                .authtoken(authtoken)
-                .connect()
-                .await
-                .unwrap()
-                .labeled_tunnel()
-                .label("edge", edge);
-
-            let listener = tunnel.listen().await.unwrap();
-
-            // Run prom metrics and health locally too
-            tokio::spawn(
-                axum::Server::bind(&addr)
-                    .serve(
-                        Router::new()
-                            .route("/health", get(health))
```

Comment:
@noyoshi @magdyksaleh not sure if this is needed and we need to add this back? Let me know.

Reply:
yes can we please keep it

Reply:
Looks like this was only enabled when using ngrok tunneling, which is unsupported at the moment. Are we using ngrok today?